src/game/SpellEffects.cpp

```diff
@@ -3006,12 +3016,21 @@ void Spell::EffectSummonPet(SpellEffectIndex eff_idx)
         ((Player*)m_caster)->PetSpellInitialize();
     }
     else
-    {
-        // Notify Summoner
+    {   // Notify Summoner and assist if in combat
         if (m_originalCaster && m_originalCaster != m_caster && m_originalCaster->GetTypeId() == TYPEID_UNIT && ((Creature*)m_originalCaster)->AI())
```

Comment:
Hmmm, that m_originalCaster->GetTypeId() == TYPEID_UNIT seems a bit redundant here since this wholse else shabang will only execute if m_caster->GetTypeId() != TYPEID_PLAYER